common/persistence/sql/sqlExecutionStore.go

```diff
@@ -791,12 +794,23 @@ func (m *sqlExecutionStore) RangeCompleteTransferTask(
 	ctx context.Context,
 	request *p.RangeCompleteTransferTaskRequest,
 ) error {
-
-	if _, err := m.db.DeleteFromTransferTasks(ctx, &sqlplugin.TransferTasksFilter{
-		ShardID:   m.shardID,
-		MinTaskID: &request.ExclusiveBeginTaskID,
-		MaxTaskID: &request.InclusiveEndTaskID}); err != nil {
-		return convertCommonErrors(m.db, "RangeCompleteTransferTask", "", err)
+	for {
```

Comment:
It doesn't need to be done within a single transaction. It guarantees that if the function returns no error, all target rows will be deleted.